druntime/src/core/internal/atomic.d

```diff
@@ -688,39 +688,57 @@ else version (GNU)
     import gcc.builtins;
     import gcc.config;
 
+    // Targets where MemoryOrder.acq_rel is sufficiently cheaper than using
+    // MemoryOrder.seq, used when the MemoryOrder requested is not valid for
+    // a given atomic operation.
```

Comment:
What's the point in making an incorrect behaviour more complicated if it's already not valid?

Reply:
In all invalid cases, the caller isn't requesting for a globally atomic operation. No need to punish them at runtime where it hurts the most.

Reply:
BTW It may also be of note that X86 isn't sequentially consistent

Reply:
I haven't decided yet, but this might be a precursor to tightening which models certain targets use anyway. I.e allow raw/relaxed only for targets where there's a genuine cost difference between raw and seq (I don't recall x86 being one of these targets), then there's this which prefers acq_rel models if you didn't explicitly request seq, for everyone else, you only get seq.

Reply:
> allow raw/relaxed only for targets where there's a genuine cost difference between raw and seq (I don't recall x86 being one of these targets)

X86 is one of those targets. 
![image](https://github.com/dlang/dmd/assets/5672948/dbd4545d-d703-4839-8b07-2cd033017877)
locking isn't free